### PR TITLE
Add SummaryDataPoint support to Metrics proto (opentelemetry/opentele…

### PR DESCRIPTION
**Pull request by gcacace**
…metry-specification#1146)

* Add IntSummary and DoubleSummary to data
* Add IntSummaryDataPoint and DoubleSummaryDataPoint
* Comments

### CODE REVIEW

**Comment by gcacace** [comment]
Please consider the main use case for `IntSummary` and `DoubleSummary` is for transport purposes (not aggregation). In this specific case I've removed `AggregationTemporality`, addressing concerns expressed in open-telemetry/opentelemetry-proto#199.

I've added `min` and `max`.

Not sure if we need to support both `IntSummary` and `DoubleSummary`, or the Double version is enough (I've included both for model completeness).

Not sure if we need to support `exemplars`.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (JasonXZLiu) [inline opentelemetry/proto/metrics/v1/metrics.proto:222]: Following a previous discussion of the summary metric (in the metrics SIG) and [here](https://github.com/open-telemetry/opentelemetry-proto/pull/199), I believe @bogdandrutu suggested to mark this data format as Legacy. This would help avoid confusion and prevent future instruments from being built on the summary (as we're mainly using it to transport Prometheus summary metrics).

@jmacd @bogdandrutu Would you have any opinions here?

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (bogdandrutu) [inline opentelemetry/proto/metrics/v1/metrics.proto:559]: I would not add this for the moment, we can followup. The purpose of this is to be compatible with OpenMetrics/Prometheus and they don't define that.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (jmacd) [reply opentelemetry/proto/metrics/v1/metrics.proto:222]: I'll accept "Legacy", yes.
Because of this legacy, should we only support Double summaries?
The precise wording can be:

// DoubleSummary metric data are used to convey quantile summaries, 
// a Prometheus and OpenMetrics data type. These data points cannot 
// always be merged in a meaningful way. While they can be useful in some
// applications, histogram data points are recommended for new applications.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (gcacace) [reply opentelemetry/proto/metrics/v1/metrics.proto]: I've removed them, thanks for the comment

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (gcacace) [reply opentelemetry/proto/metrics/v1/metrics.proto]: I've removed `IntSummary` and added the suggested comment the `DoubleSummary`.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (gcacace) [inline opentelemetry/proto/metrics/v1/metrics.proto:576]: I'm wondering if we need also exemplars or we can get rid of them

Comment (gcacace) [reply opentelemetry/proto/metrics/v1/metrics.proto:576]: I'm gonna get rid of that since it's not required for Prometheus use case.

**Review by jmacd — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Review by JasonXZLiu — APPROVED** [review]

**Comment by gcacace** [comment]
Can someone with merge permissions push and close this change please? When it is going to be released in Maven?

**Comment by bogdandrutu** [comment]
We don't release this repo to maven (that is a Java specific repo, if I am not wrong). The other projects consume this repo as git submodule and generates their code.

**Comment by bogdandrutu** [comment]
@open-telemetry/specs-metrics-approvers please take a look.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (rakyll) [inline opentelemetry/proto/metrics/v1/metrics.proto:225]: Prometheus doesn't have support for int summaries but I'm seeing OTLP represents int histograms. Are we planning to do the same here? If yes, it would be good to add a TODO at least.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (rakyll) [inline opentelemetry/proto/metrics/v1/metrics.proto:572]: Does Prometheus summary type support exemplars yet?

**Review by rakyll — COMMENTED** [review]
I'm not familiar with the OTLP yet to give a more comprehensive review but I am leaving two high level questions.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:572]: Not that I know of

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:225]: We are not planning to support the int version since this is mostly for backwards compatibility with Prometheus

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (JasonXZLiu) [reply opentelemetry/proto/metrics/v1/metrics.proto:572]: Looking at the [Prometheus client proto](https://github.com/prometheus/client_model/blob/master/go/metrics.pb.go), there is no support for exemplars with the Prometheus Summary type. (It's only available for Counters and Histograms.)

Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:572]: If they add that we will also add it.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:225]: If they add that we will add it

**Comment by gcacace** [comment]
> We don't release this repo to maven (that is a Java specific repo, if I am not wrong). The other projects consume this repo as git submodule and generates their code.

Who should I ask in order to have a released version on this code in Maven then? https://mvnrepository.com/artifact/io.opentelemetry/opentelemetry-proto

**Comment by bogdandrutu** [comment]
@gcacace that is generated by the opentelemetry-java repo. So once this is merged, and the otel java repo resyncs will export this to maven.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (MrAlias) [inline opentelemetry/proto/metrics/v1/metrics.proto:520]:
```suggestion
// DoubleSummaryDataPoint is a single data point in a timeseries that describes the
```

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (MrAlias) [inline opentelemetry/proto/metrics/v1/metrics.proto:222]: Should we include that these are used to also record `MinMaxSumCount` values?

**Review by MrAlias — APPROVED** [review]

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (gcacace) [reply opentelemetry/proto/metrics/v1/metrics.proto:520]: I'm following the same convention on comments of the other data structures

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (JasonXZLiu) [reply opentelemetry/proto/metrics/v1/metrics.proto:222]: I'm not too sure, but I believe there wasn't a final decision on whether the DoubleSummary would be used to record MinMaxSumCount values. I believe it's only used for backwards compatibility with the Prometheus Summary metric for now.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (tigrannajaryan) [reply opentelemetry/proto/metrics/v1/metrics.proto:520]: I see 
```
// IntDataPoint is a single data point in a timeseries that describes the
// time-varying values of a int64 metric.
message IntDataPoint {
```
and
```
// DoubleDataPoint is a single data point in a timeseries that describes the
// time-varying value of a double metric.
message DoubleDataPoint {
```
The convention should be to use the full name of the message in the comment. If elsewhere it is not so it needs to be fixed too. DoubleHistogramDataPoint appears to be violating the convention, can be fixed in a future PR.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (tigrannajaryan) [inline opentelemetry/proto/metrics/v1/metrics.proto:547]: uint64 is possibly a better choice since counts tend to be relatively small numbers which will pack better on the wire. I see histogram does this too, so it is probably best to change it in a separate PR if we want to do it.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (tigrannajaryan) [inline opentelemetry/proto/metrics/v1/metrics.proto:222]: Can we leave out the references to Prometheus and OpenMetrics? Do they add anything of value? If it is important for understanding the semantics then it is best to include a link to Prometheus and OpenMetrics that explain it.

**Comment by gcacace** [comment]
How can I get more traction on closing out this PR?

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:222]: Yes, @MrAlias we decided to postpone that decision for the moment.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:547]: We had this discussion before. Let's keep fixed64 and rethink this everywhere instead of having an inconsistency

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:222]: Making this a link is better. We want to keep the reference because as the PR explains the summary is mostly to be able to carry openmetrics/Prometheus summaries not for us to produce them

**Comment by bogdandrutu** [comment]
@gcacace i think the author needs to respond and fix some of the comments.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (gcacace) [reply opentelemetry/proto/metrics/v1/metrics.proto:222]: Just to get some clarity on the ask, do you want me to add links to https://prometheus.io/ and https://openmetrics.io/ ?

Comment (tigrannajaryan) [reply opentelemetry/proto/metrics/v1/metrics.proto:222]: Yes, the links would be very useful. Also add a relevant OpenMetrics link.

**Comment by gcacace** [comment]
All comments addressed.

**Review by tigrannajaryan — APPROVED** [review]
LGTM

**Comment by JasonXZLiu** [comment]
Would it be possible to make a release with these changes right after merging this PR? We have some blocked [PR](https://github.com/open-telemetry/opentelemetry-collector/pull/2048)'s waiting for this datapoint to be added to the proto.

**Comment by bogdandrutu** [comment]
Don't need a release just update the submodule to point to the merged commit

**Pull request merged**
